### PR TITLE
fix for #518

### PR DESCRIPTION
**Pull request by osa1**
#518

### CODE REVIEW

**Comment by osa1** [comment]
Failed because of timeout.

**Comment by dwightguth** [comment]
Jenkins: test this please

File: src/javasources/KTool/src/org/kframework/utils/StringUtil.java [hunk 1]
```diff
@@ -525,19 +525,25 @@ private static String splitLine(String str, int col) {
     public static String escapeShell(String[] args, OS os) {
         if (os.isPosix) {
             StringBuilder sb = new StringBuilder();
-            for (String arg : args) {
-                sb.append("'");
+            for (int i = 0; i < args.length; i++) {
+                String arg = args[i];
+                // Don't quote first argument since it's program name to run and quoting it
+                // makes the command invalid to run
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/utils/StringUtil.java]: says who? works for me...

File: src/javasources/KTool/src/org/kframework/utils/StringUtil.java [hunk 2]
```diff
@@ -525,19 +525,25 @@ private static String splitLine(String str, int col) {
...
+                if (i != 0) sb.append('\'');
+                sb.append(' ');
             }
             sb.deleteCharAt(sb.length() - 1);
             return sb.toString();
         } else if (os == OS.WIN) {
             StringBuilder sb = new StringBuilder();
-            for (String arg : args) {
-                sb.append('"');
+            for (int i = 0; i < args.length; i++) {
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/utils/StringUtil.java]: okay, I know what the problem is here. We are escaping backslashes but in fact windows doesn't escape backslashes that way. Windows escapes quotes with backslash but doesn't escape backslash. The correct solution is to escape \ into \ but to still escape " into \".

**Comment by dwightguth** [comment]
make sure you read my comment on the windows code

File: src/javasources/KTool/src/org/kframework/utils/StringUtil.java [hunk 3]
```diff
@@ -536,7 +536,7 @@ public static String escapeShell(String[] args, OS os) {
             StringBuilder sb = new StringBuilder();
             for (String arg : args) {
                 sb.append('"');
-                sb.append(StringUtils.replace(StringUtils.replace(arg, "\\", "\\\\"), "\"", "\\\""));
+                sb.append(StringUtils.replace(arg, "\"", "\\\""));
```
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/utils/StringUtil.java:539]: do you mean this?

**Pull request merged**